diff --git a/src/base/SkArenaAlloc.h b/src/base/SkArenaAlloc.h
index fd3e652daf..0c85e7f114 100644
--- a/src/base/SkArenaAlloc.h
+++ b/src/base/SkArenaAlloc.h
@@ -103,7 +103,7 @@ private:
 // recursion of the RunDtorsOnBlock to be limited to O(log size-of-memory). Block size grow using
 // the Fibonacci sequence which means that for 2^32 memory there are 48 allocations, and for 2^48
 // there are 71 allocations.
-class SkArenaAlloc {
+class SK_API SkArenaAlloc {
 public:
     SkArenaAlloc(char* block, size_t blockSize, size_t firstHeapAllocation);
 
diff --git a/src/base/SkBezierCurves.h b/src/base/SkBezierCurves.h
index 581c03277c..987aeab2e7 100644
--- a/src/base/SkBezierCurves.h
+++ b/src/base/SkBezierCurves.h
@@ -24,7 +24,7 @@ struct SkPoint;
  * the even indices are the X coordinates, and the odd indices are the Y
  * coordinates.
  */
-class SkBezierCubic {
+class SK_API SkBezierCubic {
 public:
 
     /**
diff --git a/src/base/SkCubics.h b/src/base/SkCubics.h
index acc4be6156..cd88ce5fca 100644
--- a/src/base/SkCubics.h
+++ b/src/base/SkCubics.h
@@ -7,13 +7,15 @@
 #ifndef SkCubics_DEFINED
 #define SkCubics_DEFINED
 
+#include "include/private/base/SkAPI.h"
+
 #include <cmath>
 
 /**
  * Utilities for dealing with cubic formulas with one variable:
  *   f(t) = A*t^3 + B*t^2 + C*t + d
  */
-class SkCubics {
+class SK_API SkCubics {
 public:
     /**
      * Puts up to 3 real solutions to the equation
diff --git a/src/core/SkGeometry.h b/src/core/SkGeometry.h
index d0582e0ff9..372cf3e1cc 100644
--- a/src/core/SkGeometry.h
+++ b/src/core/SkGeometry.h
@@ -36,37 +36,37 @@ static skvx::float2 times_2(const skvx::float2& value) {
 /** Given a quadratic equation Ax^2 + Bx + C = 0, return 0, 1, 2 roots for the
     equation.
 */
-int SkFindUnitQuadRoots(SkScalar A, SkScalar B, SkScalar C, SkScalar roots[2]);
+SK_API int SkFindUnitQuadRoots(SkScalar A, SkScalar B, SkScalar C, SkScalar roots[2]);
 
 /** Measures the angle between two vectors, in the range [0, pi].
 */
-float SkMeasureAngleBetweenVectors(SkVector, SkVector);
+SK_API float SkMeasureAngleBetweenVectors(SkVector, SkVector);
 
 /** Returns a new, arbitrarily scaled vector that bisects the given vectors. The returned bisector
     will always point toward the interior of the provided vectors.
 */
-SkVector SkFindBisector(SkVector, SkVector);
+SK_API SkVector SkFindBisector(SkVector, SkVector);
 
 ///////////////////////////////////////////////////////////////////////////////
 
-SkPoint SkEvalQuadAt(const SkPoint src[3], SkScalar t);
-SkPoint SkEvalQuadTangentAt(const SkPoint src[3], SkScalar t);
+SK_API SkPoint SkEvalQuadAt(const SkPoint src[3], SkScalar t);
+SK_API SkPoint SkEvalQuadTangentAt(const SkPoint src[3], SkScalar t);
 
 /** Set pt to the point on the src quadratic specified by t. t must be
     0 <= t <= 1.0
 */
-void SkEvalQuadAt(const SkPoint src[3], SkScalar t, SkPoint* pt, SkVector* tangent = nullptr);
+SK_API void SkEvalQuadAt(const SkPoint src[3], SkScalar t, SkPoint* pt, SkVector* tangent = nullptr);
 
 /** Given a src quadratic bezier, chop it at the specified t value,
     where 0 < t < 1, and return the two new quadratics in dst:
     dst[0..2] and dst[2..4]
 */
-void SkChopQuadAt(const SkPoint src[3], SkPoint dst[5], SkScalar t);
+SK_API void SkChopQuadAt(const SkPoint src[3], SkPoint dst[5], SkScalar t);
 
 /** Given a src quadratic bezier, chop it at the specified t == 1/2,
     The new quads are returned in dst[0..2] and dst[2..4]
 */
-void SkChopQuadAtHalf(const SkPoint src[3], SkPoint dst[5]);
+SK_API void SkChopQuadAtHalf(const SkPoint src[3], SkPoint dst[5]);
 
 /** Measures the rotation of the given quadratic curve in radians.
 
@@ -83,7 +83,7 @@ inline float SkMeasureQuadRotation(const SkPoint pts[3]) {
 /** Given a src quadratic bezier, returns the T value whose tangent angle is halfway between the
     tangents at p0 and p3.
 */
-float SkFindQuadMidTangent(const SkPoint src[3]);
+SK_API float SkFindQuadMidTangent(const SkPoint src[3]);
 
 /** Given a src quadratic bezier, chop it at the tangent whose angle is halfway between the
     tangents at p0 and p2. The new quads are returned in dst[0..2] and dst[2..4].
@@ -100,7 +100,7 @@ inline void SkChopQuadAtMidTangent(const SkPoint src[3], SkPoint dst[5]) {
     0                   ignored
     1                   0 < tValues[0] < 1
 */
-int SkFindQuadExtrema(SkScalar a, SkScalar b, SkScalar c, SkScalar tValues[1]);
+SK_API int SkFindQuadExtrema(SkScalar a, SkScalar b, SkScalar c, SkScalar tValues[1]);
 
 /** Given 3 points on a quadratic bezier, chop it into 1, 2 beziers such that
     the resulting beziers are monotonic in Y. This is called by the scan converter.
@@ -108,14 +108,14 @@ int SkFindQuadExtrema(SkScalar a, SkScalar b, SkScalar c, SkScalar tValues[1]);
     0   dst[0..2] is the original quad
     1   dst[0..2] and dst[2..4] are the two new quads
 */
-int SkChopQuadAtYExtrema(const SkPoint src[3], SkPoint dst[5]);
-int SkChopQuadAtXExtrema(const SkPoint src[3], SkPoint dst[5]);
+SK_API int SkChopQuadAtYExtrema(const SkPoint src[3], SkPoint dst[5]);
+SK_API int SkChopQuadAtXExtrema(const SkPoint src[3], SkPoint dst[5]);
 
 /** Given 3 points on a quadratic bezier, if the point of maximum
     curvature exists on the segment, returns the t value for this
     point along the curve. Otherwise it will return a value of 0.
 */
-SkScalar SkFindQuadMaxCurvature(const SkPoint src[3]);
+SK_API SkScalar SkFindQuadMaxCurvature(const SkPoint src[3]);
 
 /** Given 3 points on a quadratic bezier, divide it into 2 quadratics
     if the point of maximum curvature exists on the quad segment.
@@ -124,45 +124,45 @@ SkScalar SkFindQuadMaxCurvature(const SkPoint src[3]);
     2   dst[0..2] and dst[2..4] are the two new quads
     If dst == null, it is ignored and only the count is returned.
 */
-int SkChopQuadAtMaxCurvature(const SkPoint src[3], SkPoint dst[5]);
+SK_API int SkChopQuadAtMaxCurvature(const SkPoint src[3], SkPoint dst[5]);
 
 /** Given 3 points on a quadratic bezier, use degree elevation to
     convert it into the cubic fitting the same curve. The new cubic
     curve is returned in dst[0..3].
 */
-void SkConvertQuadToCubic(const SkPoint src[3], SkPoint dst[4]);
+SK_API void SkConvertQuadToCubic(const SkPoint src[3], SkPoint dst[4]);
 
 ///////////////////////////////////////////////////////////////////////////////
 
 /** Set pt to the point on the src cubic specified by t. t must be
     0 <= t <= 1.0
 */
-void SkEvalCubicAt(const SkPoint src[4], SkScalar t, SkPoint* locOrNull,
+SK_API void SkEvalCubicAt(const SkPoint src[4], SkScalar t, SkPoint* locOrNull,
                    SkVector* tangentOrNull, SkVector* curvatureOrNull);
 
 /** Given a src cubic bezier, chop it at the specified t value,
     where 0 <= t <= 1, and return the two new cubics in dst:
     dst[0..3] and dst[3..6]
 */
-void SkChopCubicAt(const SkPoint src[4], SkPoint dst[7], SkScalar t);
+SK_API void SkChopCubicAt(const SkPoint src[4], SkPoint dst[7], SkScalar t);
 
 /** Given a src cubic bezier, chop it at the specified t0 and t1 values,
     where 0 <= t0 <= t1 <= 1, and return the three new cubics in dst:
     dst[0..3], dst[3..6], and dst[6..9]
 */
-void SkChopCubicAt(const SkPoint src[4], SkPoint dst[10], float t0, float t1);
+SK_API void SkChopCubicAt(const SkPoint src[4], SkPoint dst[10], float t0, float t1);
 
 /** Given a src cubic bezier, chop it at the specified t values,
     where 0 <= t0 <= t1 <= ... <= 1, and return the new cubics in dst:
     dst[0..3],dst[3..6],...,dst[3*t_count..3*(t_count+1)]
 */
-void SkChopCubicAt(const SkPoint src[4], SkPoint dst[], const SkScalar t[],
+SK_API void SkChopCubicAt(const SkPoint src[4], SkPoint dst[], const SkScalar t[],
                    int t_count);
 
 /** Given a src cubic bezier, chop it at the specified t == 1/2,
     The new cubics are returned in dst[0..3] and dst[3..6]
 */
-void SkChopCubicAtHalf(const SkPoint src[4], SkPoint dst[7]);
+SK_API void SkChopCubicAtHalf(const SkPoint src[4], SkPoint dst[7]);
 
 /** Given a cubic curve with no inflection points, this method measures the rotation in radians.
 
@@ -177,12 +177,12 @@ void SkChopCubicAtHalf(const SkPoint src[4], SkPoint dst[7]);
     NOTE: The caller must either call SkChopCubicAtInflections or otherwise prove that the provided
     cubic has no inflection points prior to calling this method.
 */
-float SkMeasureNonInflectCubicRotation(const SkPoint[4]);
+SK_API float SkMeasureNonInflectCubicRotation(const SkPoint[4]);
 
 /** Given a src cubic bezier, returns the T value whose tangent angle is halfway between the
     tangents at p0 and p3.
 */
-float SkFindCubicMidTangent(const SkPoint src[4]);
+SK_API float SkFindCubicMidTangent(const SkPoint src[4]);
 
 /** Given a src cubic bezier, chop it at the tangent whose angle is halfway between the
     tangents at p0 and p3. The new cubics are returned in dst[0..3] and dst[3..6].
@@ -205,7 +205,7 @@ inline void SkChopCubicAtMidTangent(const SkPoint src[4], SkPoint dst[7]) {
     1                   0 < tValues[0] < 1
     2                   0 < tValues[0] < tValues[1] < 1
 */
-int SkFindCubicExtrema(SkScalar a, SkScalar b, SkScalar c, SkScalar d,
+SK_API int SkFindCubicExtrema(SkScalar a, SkScalar b, SkScalar c, SkScalar d,
                        SkScalar tValues[2]);
 
 /** Given 4 points on a cubic bezier, chop it into 1, 2, 3 beziers such that
@@ -216,26 +216,26 @@ int SkFindCubicExtrema(SkScalar a, SkScalar b, SkScalar c, SkScalar d,
     2   dst[0..3], dst[3..6], dst[6..9] are the three new cubics
     If dst == null, it is ignored and only the count is returned.
 */
-int SkChopCubicAtYExtrema(const SkPoint src[4], SkPoint dst[10]);
-int SkChopCubicAtXExtrema(const SkPoint src[4], SkPoint dst[10]);
+SK_API int SkChopCubicAtYExtrema(const SkPoint src[4], SkPoint dst[10]);
+SK_API int SkChopCubicAtXExtrema(const SkPoint src[4], SkPoint dst[10]);
 
 /** Given a cubic bezier, return 0, 1, or 2 t-values that represent the
     inflection points.
 */
-int SkFindCubicInflections(const SkPoint src[4], SkScalar tValues[2]);
+SK_API int SkFindCubicInflections(const SkPoint src[4], SkScalar tValues[2]);
 
 /** Return 1 for no chop, 2 for having chopped the cubic at a single
     inflection point, 3 for having chopped at 2 inflection points.
     dst will hold the resulting 1, 2, or 3 cubics.
 */
-int SkChopCubicAtInflections(const SkPoint src[4], SkPoint dst[10]);
+SK_API int SkChopCubicAtInflections(const SkPoint src[4], SkPoint dst[10]);
 
-int SkFindCubicMaxCurvature(const SkPoint src[4], SkScalar tValues[3]);
-int SkChopCubicAtMaxCurvature(const SkPoint src[4], SkPoint dst[13],
+SK_API int SkFindCubicMaxCurvature(const SkPoint src[4], SkScalar tValues[3]);
+SK_API int SkChopCubicAtMaxCurvature(const SkPoint src[4], SkPoint dst[13],
                               SkScalar tValues[3] = nullptr);
 /** Returns t value of cusp if cubic has one; returns -1 otherwise.
  */
-SkScalar SkFindCubicCusp(const SkPoint src[4]);
+SK_API SkScalar SkFindCubicCusp(const SkPoint src[4]);
 
 /** Given a monotonically increasing or decreasing cubic bezier src, chop it
  *  where the X value is the specified value. The returned cubics will be in
@@ -247,7 +247,7 @@ SkScalar SkFindCubicCusp(const SkPoint src[4]);
  *
  *  If the cubic never reaches the specified value, the function returns false.
 */
-bool SkChopMonoCubicAtX(const SkPoint src[4], SkScalar x, SkPoint dst[7]);
+SK_API bool SkChopMonoCubicAtX(const SkPoint src[4], SkScalar x, SkPoint dst[7]);
 
 /** Given a monotonically increasing or decreasing cubic bezier src, chop it
  *  where the Y value is the specified value. The returned cubics will be in
@@ -259,7 +259,7 @@ bool SkChopMonoCubicAtX(const SkPoint src[4], SkScalar x, SkPoint dst[7]);
  *
  *  If the cubic never reaches the specified value, the function returns false.
 */
-bool SkChopMonoCubicAtY(const SkPoint src[4], SkScalar y, SkPoint dst[7]);
+SK_API bool SkChopMonoCubicAtY(const SkPoint src[4], SkScalar y, SkPoint dst[7]);
 
 enum class SkCubicType {
     kSerpentine,
@@ -313,7 +313,7 @@ static inline const char* SkCubicTypeName(SkCubicType type) {
 
     https://www.microsoft.com/en-us/research/wp-content/uploads/2005/01/p1000-loop.pdf
 */
-SkCubicType SkClassifyCubic(const SkPoint p[4], double t[2] = nullptr, double s[2] = nullptr,
+SK_API SkCubicType SkClassifyCubic(const SkPoint p[4], double t[2] = nullptr, double s[2] = nullptr,
                             double d[4] = nullptr);
 
 ///////////////////////////////////////////////////////////////////////////////
@@ -323,7 +323,7 @@ enum SkRotationDirection {
     kCCW_SkRotationDirection
 };
 
-struct SkConic {
+struct SK_API SkConic {
     SkConic() {}
     SkConic(const SkPoint& p0, const SkPoint& p1, const SkPoint& p2, SkScalar w) {
         this->set(p0, p1, p2, w);
@@ -420,7 +420,7 @@ namespace {  // NOLINT(google-build-namespaces)
 /**
  *  use for : eval(t) == A * t^2 + B * t + C
  */
-struct SkQuadCoeff {
+struct SK_API SkQuadCoeff {
     SkQuadCoeff() {}
 
     SkQuadCoeff(const skvx::float2& A, const skvx::float2& B, const skvx::float2& C)
@@ -447,7 +447,7 @@ struct SkQuadCoeff {
     skvx::float2 fC;
 };
 
-struct SkConicCoeff {
+struct SK_API SkConicCoeff {
     SkConicCoeff(const SkConic& conic) {
         skvx::float2 p0 = from_point(conic.fPts[0]);
         skvx::float2 p1 = from_point(conic.fPts[1]);
@@ -475,7 +475,7 @@ struct SkConicCoeff {
     SkQuadCoeff fDenom;
 };
 
-struct SkCubicCoeff {
+struct SK_API SkCubicCoeff {
     SkCubicCoeff(const SkPoint src[4]) {
         skvx::float2 P0 = from_point(src[0]);
         skvx::float2 P1 = from_point(src[1]);
@@ -505,7 +505,7 @@ struct SkCubicCoeff {
 /**
  *  Help class to allocate storage for approximating a conic with N quads.
  */
-class SkAutoConicToQuads {
+class SK_API SkAutoConicToQuads {
 public:
     SkAutoConicToQuads() : fQuadCount(0) {}
 
diff --git a/src/core/SkPathPriv.h b/src/core/SkPathPriv.h
index 0597b955f7..5e4c1bf2bb 100644
--- a/src/core/SkPathPriv.h
+++ b/src/core/SkPathPriv.h
@@ -45,7 +45,7 @@ struct SkPathVerbAnalysis {
     bool     valid;
 };
 
-class SkPathPriv {
+class SK_API SkPathPriv {
 public:
     static SkPathVerbAnalysis AnalyzeVerbs(SkSpan<const uint8_t> verbs);
 
