diff --git a/src/libANGLE/renderer/vulkan/vulkan_backend.gni b/src/libANGLE/renderer/vulkan/vulkan_backend.gni
index 717b1ff9c5..3a5bc4f256 100644
--- a/src/libANGLE/renderer/vulkan/vulkan_backend.gni
+++ b/src/libANGLE/renderer/vulkan/vulkan_backend.gni
@@ -142,27 +142,25 @@ if (angle_use_vulkan_null_display) {
   ]
 }
 
-vulkan_backend_dma_buf_sources = [
-  "linux/DeviceVkLinux.cpp",
-  "linux/DeviceVkLinux.h",
-  "linux/DisplayVkLinux.cpp",
-  "linux/DisplayVkLinux.h",
-  "linux/DisplayVkOffscreen.cpp",
-  "linux/DisplayVkOffscreen.h",
-  "linux/DmaBufImageSiblingVkLinux.cpp",
-  "linux/DmaBufImageSiblingVkLinux.h",
-  "linux/display/DisplayVkSimple.cpp",
-  "linux/display/DisplayVkSimple.h",
-  "linux/display/WindowSurfaceVkSimple.cpp",
-  "linux/display/WindowSurfaceVkSimple.h",
-  "linux/headless/DisplayVkHeadless.cpp",
-  "linux/headless/DisplayVkHeadless.h",
-  "linux/headless/WindowSurfaceVkHeadless.cpp",
-  "linux/headless/WindowSurfaceVkHeadless.h",
-]
-
 if (is_linux || is_chromeos) {
-  vulkan_backend_sources += vulkan_backend_dma_buf_sources
+  vulkan_backend_sources += [
+    "linux/DeviceVkLinux.cpp",
+    "linux/DeviceVkLinux.h",
+    "linux/DisplayVkLinux.cpp",
+    "linux/DisplayVkLinux.h",
+    "linux/DisplayVkOffscreen.cpp",
+    "linux/DisplayVkOffscreen.h",
+    "linux/DmaBufImageSiblingVkLinux.cpp",
+    "linux/DmaBufImageSiblingVkLinux.h",
+    "linux/display/DisplayVkSimple.cpp",
+    "linux/display/DisplayVkSimple.h",
+    "linux/display/WindowSurfaceVkSimple.cpp",
+    "linux/display/WindowSurfaceVkSimple.h",
+    "linux/headless/DisplayVkHeadless.cpp",
+    "linux/headless/DisplayVkHeadless.h",
+    "linux/headless/WindowSurfaceVkHeadless.cpp",
+    "linux/headless/WindowSurfaceVkHeadless.h",
+  ]
 }
 
 if (is_android) {
