diff --git a/src/libANGLE/renderer/vulkan/vk_cache_utils.cpp b/src/libANGLE/renderer/vulkan/vk_cache_utils.cpp
index 974657a311..5a6a085177 100644
--- a/src/libANGLE/renderer/vulkan/vk_cache_utils.cpp
+++ b/src/libANGLE/renderer/vulkan/vk_cache_utils.cpp
@@ -7334,34 +7334,6 @@ UpdateDescriptorSetsBuilder::UpdateDescriptorSetsBuilder()
 
 UpdateDescriptorSetsBuilder::~UpdateDescriptorSetsBuilder() = default;
 
-template <typename T>
-T *UpdateDescriptorSetsBuilder::DescriptorInfoAllocator<T>::allocate(uint32_t count)
-{
-    size_t oldSize = mCurrentVector->size();
-    size_t newSize = oldSize + count;
-    if (newSize <= mCurrentVector->capacity())
-    {
-        (*mCurrentVector).resize(newSize);
-        mTotalSize += count;
-        return &(*mCurrentVector)[oldSize];
-    }
-
-    ++mCurrentVector;
-    // clear() always ensures we have a single element left.
-    ASSERT(mCurrentVector == mDescriptorInfos.end());
-
-    // We have reached capacity, grow the storage
-    mVectorCapacity = std::max(count, mVectorCapacity);
-    mDescriptorInfos.emplace_back();
-    mDescriptorInfos.back().reserve(mVectorCapacity);
-    mCurrentVector = mDescriptorInfos.end() - 1;
-
-    mCurrentVector->resize(count);
-    mTotalSize += count;
-
-    return &mCurrentVector->front();
-}
-
 uint32_t UpdateDescriptorSetsBuilder::WriteDescriptorSetAllocator::updateDescriptorSets(
     VkDevice device) const
 {
diff --git a/src/libANGLE/renderer/vulkan/vk_cache_utils.h b/src/libANGLE/renderer/vulkan/vk_cache_utils.h
index 057c883fdf..e7f3225fc6 100644
--- a/src/libANGLE/renderer/vulkan/vk_cache_utils.h
+++ b/src/libANGLE/renderer/vulkan/vk_cache_utils.h
@@ -3038,7 +3038,33 @@ class UpdateDescriptorSetsBuilder final : angle::NonCopyable
             mCurrentVector = mDescriptorInfos.begin();
             mTotalSize     = 0;
         }
-        T *allocate(uint32_t count);
+
+        T *allocate(uint32_t count)
+        {
+            size_t oldSize = mCurrentVector->size();
+            size_t newSize = oldSize + count;
+            if (newSize <= mCurrentVector->capacity())
+            {
+                (*mCurrentVector).resize(newSize);
+                mTotalSize += count;
+                return &(*mCurrentVector)[oldSize];
+            }
+
+            ++mCurrentVector;
+            // clear() always ensures we have a single element left.
+            ASSERT(mCurrentVector == mDescriptorInfos.end());
+
+            // We have reached capacity, grow the storage
+            mVectorCapacity = std::max(count, mVectorCapacity);
+            mDescriptorInfos.emplace_back();
+            mDescriptorInfos.back().reserve(mVectorCapacity);
+            mCurrentVector = mDescriptorInfos.end() - 1;
+
+            mCurrentVector->resize(count);
+            mTotalSize += count;
+
+            return &mCurrentVector->front();
+        }
 
         bool empty() const { return mTotalSize == 0; }
 
